### PR TITLE
SQL queries added

### PR DESCRIPTION
**Pull request by tauren44**

### CODE REVIEW

**Comment by tauren44** [comment]
do I move in the right direction making CRUD app?

File: src/main/java/com/mateacademy/Crud.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class Crud {
+    private static Logger logger = Logger.getLogger(Crud.class);
+
+    public void createData(String tableName, String values) {
+        Connection connection = Connector.getConnection();
+        Statement statement;
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: It is better to use prepared statements.
https://metanit.com/java/database/2.6.php

File: src/main/java/com/mateacademy/Crud.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        } catch (SQLException e) {
+            logger.error("SQLException: " + e);
+        }
+    }
+
+    public void readData(String tableName) throws SQLException {
+        Connection connection = Connector.getConnection();
+        ResultSet resultSet;
+        String query = "SELECT * FROM " + tableName;
+        Statement statement = connection.prepareStatement(query);
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: The same

File: src/main/java/com/mateacademy/Crud.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class Crud {
+    private static Logger logger = Logger.getLogger(Crud.class);
+
+    public void createData(String tableName, String values) {
+        Connection connection = Connector.getConnection();
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: Don't forget to close a connection.

File: src/main/java/com/mateacademy/Crud.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        String query = "SELECT * FROM " + tableName;
+        Statement statement = connection.prepareStatement(query);
+        resultSet = statement.executeQuery(query);
+        ResultSetMetaData rsmd = resultSet.getMetaData();
+        int columnsNumber = rsmd.getColumnCount();
+
+        for (int i = 1; i <= columnsNumber; i++) {
+            logger.info(rsmd.getColumnName(i) + " ");
+        }
+        logger.info("");
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: Senseless use of the logger.

File: src/main/java/com/mateacademy/Crud.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        while (resultSet.next()) {
+            for (int i = 1; i <= columnsNumber; i++) {
+                logger.info(resultSet.getString(i) + " ");
+            }
+            logger.info("");
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: Senseless use of the logger

File: src/main/java/com/mateacademy/Crud.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+    }
+
+    public void updateData(String tableName, String column, String newValue, String condition) {
+        Connection connection = Connector.getConnection();
+        Statement statement;
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: The same.

File: src/main/java/com/mateacademy/Crud.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+            statement.executeUpdate(query);
+        } catch (SQLException e) {
+            logger.error("SQLException: " + e);
+        }
+    }
+
+    public void deleteData(String tableName, String condition) {
+        Connection connection = Connector.getConnection();
+        Statement statement;
+        String query = "DELETE FROM " + tableName + " WHERE " + condition;
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: The same

File: src/main/resources/sql/task2.1.sql [hunk 8]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE developers ADD COLUMN salary INT;
+
+UPDATE developers SET salary=1700 WHERE id = 1;
+UPDATE developers SET salary=1300 WHERE id = 2;
+UPDATE developers SET salary=1200 WHERE id = 3;
+UPDATE developers SET salary=2000 WHERE id = 4;
+UPDATE developers SET salary=1700 WHERE id = 5;
+UPDATE developers SET salary=1600 WHERE id = 6;
```
Comment (Anderwerty) [inline src/main/resources/sql/task2.1.sql]: Add a blank line at the end of the file.

File: src/main/resources/sql/task2.3.sql [hunk 9]
```diff
@@ -0,0 +1,7 @@
+SELECT SUM(developers.salary)
+FROM developers_skills
+LEFT JOIN developers
+ON developers.id = developers_skills.dev_id
+LEFT JOIN skills
+ON skills.id = developers_skills.dev_id
+WHERE `skills`.skill = 'Java';
```
Comment (Anderwerty) [inline src/main/resources/sql/task2.3.sql]: Add a blank line at the end of the file.

File: src/main/resources/sql/task2.5.sql [hunk 10]
```diff
@@ -0,0 +1 @@
+SELECT MIN(cost) FROM projects;
```
Comment (Anderwerty) [inline src/main/resources/sql/task2.5.sql]: The same

File: src/main/resources/sql/task2.6.sql [hunk 11]
```diff
@@ -0,0 +1,10 @@
+SELECT AVG(developers.salary), projects.name
+FROM developers_projects
+LEFT JOIN developers
+ON developers.id = developers_projects.dev_id
+LEFT JOIN projects
+ON projects.id = developers_projects.project_id
+GROUP BY projects.id
+HAVING MIN(projects.cost)
+ORDER BY SUM(developers.salary) DESC
+LIMIT 1;
```
Comment (Anderwerty) [inline src/main/resources/sql/task2.6.sql]: The same

File: src/test/java/com/mateacademy/CrudTest.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    public void shouldDeleteDataFromChosenTableIfCondition() {
+        crud.deleteData("developers", "id = 15");
+        try {
+            crud.readData("developers");
+        } catch (SQLException e) {
+            logger.error(e);
+        }
+    }
+}
```
Comment (Anderwerty) [inline src/test/java/com/mateacademy/CrudTest.java]: The same

File: src/main/resources/sql/homework_init.sql [hunk 13]
```diff
@@ -0,0 +1,130 @@
+CREATE SCHEMA `homework` ;
```
Comment (Anderwerty) [inline src/main/resources/sql/homework_init.sql]: Better first create all tables and then insert data.

File: src/main/resources/sql/homework_init.sql [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+USE homework;
+
+CREATE TABLE `homework`.`developers` (`id` INT NOT NULL AUTO_INCREMENT,
```
Comment (Anderwerty) [inline src/main/resources/sql/homework_init.sql]: Please use the same format as below

File: src/main/java/com/mateacademy/Crud.java [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+        try (Connection connection = Connector.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("SQLException: " + e);
+        }
+    }
+
+    public void readData(String tableName) throws SQLException {
```
Comment (Anderwerty) [inline src/main/java/com/mateacademy/Crud.java]: The same. Repository per entity. 
```
public List<User> findAllUsers()
{
...

}
```

File: src/test/java/com/mateacademy/CrudTest.java [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+
+
+
+public class CrudTest {
+    private static Logger logger = Logger.getLogger(CrudTest.class);
+    private Crud crud = new Crud();
+
+    @Test
+    public void shouldCreateDataInDataBaseInChosenTable() {
+        crud.createData("developers", "'FirstName SecondName', 20, 'male', 1550");
```
Comment (Anderwerty) [inline src/test/java/com/mateacademy/CrudTest.java]: You don't have any assert in tests methods. So it is not test-method with its primary role.

File: src/test/java/com/mateacademy/CrudTest.java [hunk 17]
```diff
@@ -0,0 +1,46 @@
+package com.mateacademy;
+
+import org.apache.log4j.Logger;
+import org.junit.Test;
+import java.sql.SQLException;
+
+
+
```
Comment (Anderwerty) [inline src/test/java/com/mateacademy/CrudTest.java]: Too many blank lines.

File: src/test/java/com/mateacademy/CrudTest.java [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    @Test
+    public void shouldDeleteDataFromChosenTableIfCondition() {
+        crud.deleteData("developers", "id = 15");
+        try {
+            crud.readData("developers");
+        } catch (SQLException e) {
+            logger.error(e);
+        }
+    }
+}
```
Comment (Anderwerty) [inline src/test/java/com/mateacademy/CrudTest.java]: Add a blank line at the end of the file.

File: src/test/java/com/mateacademy/CrudTest.java (hunk 16 above)
Comment (tauren44) [reply src/test/java/com/mateacademy/CrudTest.java]: Oleksandr told that not necessary to write tests for this hw, so I created them for my own

**Pull request merged**